File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        StringBuilder sb = new StringBuilder();
        Stack<String> st = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            
            if (ch != ' ') {
                StringBuilder temp = new StringBuilder();
                
                while (i < s.length() && ch != ' ') {
                    temp.append(ch);
                    i++;  // Increment i to move to the next character
                    if (i < s.length())
                        ch = s.charAt(i);
                }
                
                st.push(temp.toString());
            }
        }

        while (!st.isEmpty()) {
            sb.append(st.pop());

            if (!st.isEmpty()) {
                sb.append(" ");
            }
        }

        return sb.toString();
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp

class Solution {
public:
#include<stack>
    string reverseWords(string s) {
        stack<string>str;
        string sb="";
        for(int i=0;i<s.length();i++){
            char ch=s[i];
            if(ch!=' '){
                string temp="";
                while(i<s.length()&&ch!=' '){
                    temp+=ch;
                    i++;
                    if(i<s.length()){
                        ch=s[i];
                    }
                }
                str.push(temp);
            }
        }
        while(!str.empty()){
            sb+=str.top();
            str.pop();
            if(!str.empty()){
                sb+=" ";
            }
        }
        return sb;

        
    }
};